vignettes/test/widgets.Rmd

```diff
@@ -15,7 +15,7 @@ Test spacing above widget.
 
 ```{r, echo=FALSE}
 # set seed for reproducible widget id
-if (requireNamespace("htmltools", quietly = TRUE)) {
+if (requireNamespace("htmlwidgets", quietly = TRUE)) {
   htmlwidgets::setWidgetIdSeed(42)
```

Comment:
Can we drop this block now?

Reply:
Nope. The htmlwidgets seed init in `R/build-reference.R` above only applies to htmlwidgets that are generated in examples (built by `build_reference()`), not vignettes/articles (built by `build_articles()`). Currently, only the former function offers a `seed` param.

I think we need to modify the `envir` passed to `rmarkdown::render()` if we also want to have stable htmlwidget IDs for articles by default:

https://github.com/r-lib/pkgdown/blob/8af84b7249634b4cc844b56aaeca8fdd209d0bc2/R/rmarkdown.R#L22

Should I give it a try? If yes, in a separate PR?